### ISSUE TITLE
JAAS Configuration overwrites Configurations created by other libraries

### ISSUE THREAD

**Issue by edanidzerda** [spring-projects/spring-kafka#2709]
**In what version(s) of Spring for Apache Kafka are you seeing this issue?**

We see this issue in 2.8.11, but I believe this has existed since [GH-201](https://github.com/spring-projects/spring-kafka/pull/310/files#)

**Describe the bug**

When utilizing Kerberos authentication with both Spring Kafka and the Microsoft SQL Server JDBC Driver, a JAAS Configuration conflict occurs depending on the order of Bean startup.  Microsoft's SQL JDBC driver creates a default Kerberos JAAS Configuration on startup, and if it starts after Spring Kafka, it will delegate requests for the `KafkaClient` configuration. 

However, if Spring Kafka initializes _after_ the SQL Datasource, the JAAS Configuration that Microsoft's SQL driver created is overwritten by [KafkaJasLoginModuleInitializer.java:127-144](https://github.com/spring-projects/spring-kafka/blob/bbefd04f953680c6346575cbf51fdcc6690a1968/spring-kafka/src/main/java/org/springframework/kafka/security/jaas/KafkaJaasLoginModuleInitializer.java#L127-L144)
```
Configuration.setConfiguration(new InternalConfiguration(configurationEntries));
```

This creates a complex to troubleshoot issue, as bean startup order determines whether or when exactly the SQL connections will begin to fail with this error:

 `No LoginModules configured for SQLJDBCDriver at javax.security.auth.login.LoginContext.init(LoginContext.java:264)`

**To Reproduce**

To reproduce this error, you need an environment with a Kerberos Domain Controller, a Microsoft SQL Server (tested with 2017 and 2019), and a Kafka installation, where both Kafka and SQL Server are configured to use Kerberos/SASL authentication.  Configure the Spring Boot datasource JDBC URL to use 'integratedSecurity=true;authenticationScheme=JavaKerberos;encrypt=true` and Configure Spring Kafka to use Kerberos.

**Expected behavior**

A Spring application configured with Spring-Kafka and a Microsoft SQL data source using Kerberos authentication should be able to coexist.  They should both be able to use their default created Kerberos configurations, and connect without issue.

**Sample**

I don't have a sample to share as the working examples are behind a corporate firewall.  This problem (or fix) doesn't require any knowledge of the corporation itself or corporate apps, it's the coexistance of Spring Kafka and mssql-jdbc.

I'd like to propose patching `InternalConfiguration` inside of KafkaJasLoginModuleInitializer to work similar to Microsoft's [`JaasConfiguration`](https://github.com/microsoft/mssql-jdbc/blob/8b3b22d0f41b9c249c52ec03123850e9d9e39cd7/src/main/java/com/microsoft/sqlserver/jdbc/JaasConfiguration.java#L51-L65):

```
    JaasConfiguration(Configuration delegate) {
        this.delegate = delegate;
        this.defaultValue = generateDefaultConfiguration();
    }


    @Override
    public AppConfigurationEntry[] getAppConfigurationEntry(String name) {
        AppConfigurationEntry[] conf = delegate == null ? null : delegate.getAppConfigurationEntry(name);
        // We return our configuration only if user requested default one
        // In case where user did request another JAAS Configuration name, we expect he knows what he is doing.
        if (conf == null && name.equals(SQLServerDriverStringProperty.JAAS_CONFIG_NAME.getDefaultValue())) {
            return defaultValue;
        }
        return conf;
    }
```

By adding a `delegate` to InternalConfiguration similar to above, the `KafkaClient `configuration would be keeping a pointer to the previous `Configuration.`  It could then delegate requests for configuration names to the previous configuration, whatever it might be, and otherwise return its own configuration entries as it does today.

Let me know if you need any additional detail.  I think the patch will be very small and easy to unit test.

**Comment by edanidzerda** [comment]
Thanks!  I agree it's an _enhancement_.  This small improvement will make it even easier for everything to "just work."

**Comment by garyrussell** [comment]
Agreed. But I haven’t had a chance to look at it yet, will do so early next week.

**Comment by garyrussell** [comment]
@caematos Disable it in what way? It should only be present if you define it as a bean, or if you enable it in Boot via https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html#application-properties.integration.spring.kafka.jaas.enabled

**Comment by sujayn105** [comment]
i faced same issue in spring cloud stream  with kafka binder.
Spring boot version: 2.7.15
spring-cloud.version: 2021.0.8

**Comment by sujayn105** [comment]
> This is implemented in 3.0.9, used by Boot 3.0.x and compatible Spring Cloud versions.

Thank you :) , Unfortunately I cannot upgrade from java 11 to make use of Boot 3.0.x. Thank you for quick reply.
